File: example1/src/main/java/com/example/grpc/features/nameresolve/ExampleNameResolverProvider.java
package com.example.grpc.features.nameresolve;

import io.grpc.NameResolver;
import io.grpc.NameResolverProvider;

import java.net.URI;

public class ExampleNameResolverProvider extends NameResolverProvider {

    @Override
    public NameResolver newNameResolver(URI targetUri, NameResolver.Args args) {
        return new ExampleNameResolver(targetUri);
    }

    @Override
    protected boolean isAvailable() {
        return true;
    }

    @Override
    protected int priority() {
        return 5;
    }

    @Override
    public String getDefaultScheme() {
        return Settings.SCHEME;
    }
}

File: grpc-kotlin-examples/src/main/kotlin/com/example/grpc/nameresolve/ExampleNameResolverProvider.kt
package com.example.grpc.nameresolve

import io.grpc.NameResolver
import io.grpc.NameResolverProvider
import java.net.URI

internal class ExampleNameResolverProvider : NameResolverProvider() {
  override fun newNameResolver(targetUri: URI?, args: NameResolver.Args?): NameResolver {
    return ExampleNameResolver(targetUri)
  }

  override fun isAvailable(): Boolean {
    return true
  }

  override fun priority(): Int {
    return 5
  }

  override fun getDefaultScheme(): String {
    return Settings.SCHEME
  }
}
